src/common/components/menuDropdown/MenuDropdown.tsx

```diff
@@ -0,0 +1,189 @@
+import { ResizeObserver } from '@juggle/resize-observer';
+import classNames from 'classnames';
+import { ButtonProps, IconAngleDown, IconAngleUp } from 'hds-react';
+import React, { useRef, useState } from 'react';
+import mergeRefs from 'react-merge-refs';
+import useMeasure from 'react-use-measure';
+
+import useClickOrFocusOutside from '../../../hooks/useClickOrFocusOutside';
+import useDropdownKeyboardNavigation from '../../../hooks/useDropdownKeyboardNavigation';
+import useIdWithPrefix from '../../../hooks/useIdWithPrefix';
+import Button from '../button/Button';
+
+// eslint-disable-next-line import/no-named-as-default
+import Menu, { MenuPosition } from './menu/Menu';
+import styles from './menuDropdown.module.scss';
+import { MenuItemOptionProps } from './types';
+
+export type MenuDropdownProps = React.PropsWithChildren<{
+  buttonAriaLabel?: string;
+  buttonLabel: string;
+  className?: string;
+  closeOnItemClick?: boolean;
+  fixedPosition?: boolean;
+  id?: string;
+  items: MenuItemOptionProps[];
+  menuPosition?: MenuPosition;
+}>;
+
+const MenuDropdown: React.FC<MenuDropdownProps> = ({
+  buttonAriaLabel,
+  buttonLabel,
+  className,
+  closeOnItemClick,
+  fixedPosition = false,
+  id: _id,
+  items,
+  menuPosition,
+}) => {
+  const disabledIndices = items.reduce(
+    (acc: number[], item, i) => (item.disabled ? [...acc, i] : acc),
+    []
+  );
+  const [menuOpen, setMenuOpen] = useState<boolean>(false);
+
+  const [ref, menuContainerSize] = useMeasure({
+    // Detect scroll events only if menu is open to improve performance
+    scroll: fixedPosition && menuOpen,
+    polyfill: ResizeObserver,
+  });
+
+  const toggleButton = useRef<HTMLButtonElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const id = useIdWithPrefix({ id: _id, prefix: 'menu-dropdown-' });
+  const buttonId = `${id}-button`;
+  const menuId = `${id}-menu`;
+
+  const ensureMenuIsClosed = () => {
+    /* istanbul ignore else */
+    if (menuOpen) {
+      setMenuOpen(false);
+      setFocusedIndex(-1);
+    }
+  };
+
+  const ensureMenuIsOpen = () => {
+    /* istanbul ignore else */
+    if (!menuOpen) {
+      setMenuOpen(true);
+    }
+  };
+
+  const setFocusToButton = () => {
+    toggleButton.current?.focus();
+  };
+
+  const handleItemClick = () => {
+    if (closeOnItemClick) {
```

Comment:
SonarCloud is showing that these lines are uncovered by tests

Reply:
Added a test to cover this.